### PR TITLE
docs: add note about reviewing PRs in CONTRIBUTING

### PR DESCRIPTION
**Pull request by trivikr**
**What's the problem this PR addresses?**

Notifies new contributors to not approve the PRs if they're not members.
Ref: https://github.com/yarnpkg/berry/pull/4060#issuecomment-1028287189

**How did you fix it?**
Added a note in CONTRIBUTING.md

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/gatsby/content/advanced/contributing.md
Comment (paul-soporan) [inline packages/gatsby/content/advanced/contributing.md:129]: I think it would be better for this section to be the last one since it isn't something we encounter on a regular basis.

Comment (trivikr) [reply packages/gatsby/content/advanced/contributing.md:129]: This section currently put after preparing your PR section. Isn't that a better location as the document flow is talking about PRs?

Comment (paul-soporan) [reply packages/gatsby/content/advanced/contributing.md:129]: :thinking: In this case, perhaps it would make more sense to move `Writing documentation` right before `Preparing your PR to be released`

Comment (trivikr) [reply packages/gatsby/content/advanced/contributing.md:129]: I'm a new contributor and do not know why writing documentation was added in the end. Could it be because it's the not the documentation directly is written in code, but in different package gatsby?

I don't think it should block this PR though.

Comment (paul-soporan) [reply packages/gatsby/content/advanced/contributing.md:129]: (Apologies for the late reply)

> do not know why writing documentation was added in the end

Neither do I since that happened 3 years ago in https://github.com/yarnpkg/berry/pull/307 before my involvement in the project, so let's ping @arcanis for details.

> Could it be because it's the not the documentation directly is written in code, but in different package gatsby?

Perhaps, but since writing documentation is something people do more often than contributing code, I feel like it would make more sense to not have it be after a section that few people find important.

> I don't think it should block this PR though.

The ordering of the sections is still important as IMO a section as important as `Writing documentation` shouldn't be buried after `Reviewing other PRs`.

"Blocking" is a bit of a strong word in this situation as, while it would be nice to have this resolved before merging this PR, it can also be done in a follow-up PR (i.e. I won't block on this if you feel strongly about not reordering the sections yourself).

Another idea I've thought about: perhaps `Writing documentation` could instead be moved right before `Writing your feature` since there are generally more people contributing documentation than features. :thinking:

Comment (trivikr) [reply packages/gatsby/content/advanced/contributing.md:129]: I created an issue to revisit ordering of sections in CONTRIBUTING at https://github.com/yarnpkg/berry/issues/4098
The decision can be taken async by yarnpkg team.

**Review by paul-soporan — APPROVED** [review]

**Pull request merged**
